### ISSUE TITLE
[Gutenboarding]: frankenflow login

### ISSUE THREAD

**Issue by ramonjd** [Automattic/wp-calypso#40004]
Where necessary, or in order to deliver our MVP, let's consider redirecting to existing pages for:

- 2fa authentication
- Magic login link notification (?) If we don't decide to implement https://github.com/Automattic/wp-calypso/issues/40002
- Other login flows as required

For example, for users with 2fa enabled, we'll redirect to the current 2fa page from Gutenboarding login.

<img width="514" alt="Screen Shot 2020-03-10 at 3 47 06 pm" src="https://user-images.githubusercontent.com/6458278/76280572-6da8ca80-62e6-11ea-9807-00c218d8f2ca.png">

### Tasks

- [ ] P2 about this approach and link it to the Master Thread (pbxlJb-5E-p2)
- [ ] Get design input
- [ ] Audit the various login flows (pbAok1-od-p2) and decide what needs to be "Frankenflowed"

**Comment by roo2** [comment]
I think this will need to be done for Magic login link users, the problem is that we don't get their session cookie until they've followed the link from their email. But in the tab they email from , they might not have access to the gutenboarding store state with their site's name, category, etc.
We have other options to implement magic login links

1) Assume the user opened the link in the same browser session as gutenboarding and reach into the shared local storage to retrieve the gutenboarding state
2) Pass all of the gutenboarding state in the URL.
 - JSON.stringify what is in local storage and then find a way to re-hydrate it in the new tab
3) Use a server side storage method for the gutenboarding state?
4) It's easiest just to redirect them to the existing login page and not support them in gutenboarding for now

**Comment by ramonjd** [comment]
> we don't get their session cookie until they've followed the link from their email. 

dang, thanks a lot for looking into this @roo2 !

sounds like number 4 will be our go to method

**Comment by roo2** [comment]
It looks like the 2FA login flows will be reasonably straightforward to handle from within our gutenboarding page ( haven't checked them all but the basic concept of putting the page into a different state while we wait for a second factor code works)

**Comment by p-jackson** [comment]
I'm working on #40236 which frankenflows the entire login flow. Not quite what this task is about (which appears to be more about jumping into the `/log-in` flow only after we realise we need to), but related.

**Comment by ramonjd** [comment]
I'll park this issue. Might eventually close it since @p-jackson has done the MVP work in #40236 

🙇

**Comment by ramonjd** [comment]
We've taken this as far as we can for MVP. Let's create a new issue for the next iteration.
